src/serve/options.ts

```diff
@@ -35,11 +35,23 @@ export const processOptions = <TResponse extends Response = Response, TInitialPa
       baseUrl: environment.QSTASH_URL!,
       token: environment.QSTASH_TOKEN!,
     }),
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    onStepFinish: (workflowRunId: string, _finishCondition: FinishCondition) =>
-      new Response(JSON.stringify({ workflowRunId }), {
+    onStepFinish: (workflowRunId: string, finishCondition: FinishCondition) => {
+      if (finishCondition === "auth-fail") {
+        console.error(AUTH_FAIL_MESSAGE);
```

Comment:
Maybe we can log with `debug?.log`

Reply:
How about we have both? If we only have debug, then the user won't understand what's happening if they get this error uninentionally

Reply:
debug.log is async. onStepFinish would have to be async. it's a breaking change. I think we should do it later.